src/ir/graph.partition.cpp

```diff
@@ -333,22 +331,28 @@ class graph_merger
     bool check_circle(std::list<region>::iterator ita, std::list<region>::iterator itb)
     {
         // merge directly
-        if (ita->outputs.size() == 1)
+        bool merge_directly = true;
```

Comment:
If all outputs of one region are connected to another region, they should be merged.